### DISCUSSION TITLE
A simple key input to manage translation

### DISCUSSION THREAD

**Discussion by delleloper** [dialogic-godot/dialogic#822] [Ideas]
I was thinking about translations
What if you can set a translation key to a text event if translations are enabled, something like this
![image](https://user-images.githubusercontent.com/17836214/157079749-e2c33a5c-29fe-4116-b4fd-a43d9124f816.png)
it could look similar to the options in the character event
![image](https://user-images.githubusercontent.com/17836214/157079096-ae580c44-b429-4ba6-a696-b6282ecc1aea.png)

The text in the editor could be the default language, and you set the locale id of the default language in the options

**Comment by AnidemDex** [comment]
@KvaGram is working on something related to that, see https://github.com/KvaGram/dialogic/tree/multilang-support

Making translations comes in 3 core parts:
- What format will the translation use
- Where is going to be saved
- How is going to be modified

**Reply by KvaGram** [reply]
Indeed.
I got my answer to those three.
1 Stored internally with the default dialouge.
2 see above
3 in the editor.
This is my philosophy when it comes to translations. To keep it in the editor. For that reason, my implementation will allow you to write translations directly without the need for translation keys. This is great for bilingual solo indie devs like myself.

As I have discussed with @coppolaemilio , this won't work for everyone. So a supportive tool implementing translation keys for one translation standard or another are already on the planning board.

**Reply by coppolaemilio** [reply]
I think that there is great value to what @KvaGram is making and we can totally work on that to support this kind of in-editor translations. So looking forward to seeing the changes to integrate in the future!

**Comment by coppolaemilio** [comment]
Translations are a must, but I'm not sure if we will be able to add them to 1.4 or if it will be a 2.0 thing. We can keep this discussion open so we track it 👍🏻

**Comment by delleloper** [comment]
I checked @KvaGram work and it looks awesome, this is like a small improvement to the system we already have, personally i prefer having translations on a file outside of the editor.
Anyways i will try to implement it and make a PR at least for learning purposes, looks like an "easy" thing to learn how to contribute.
